core/sdk-core/src/main/java/software/amazon/awssdk/core/async/AsyncRequestBody.java

```diff
@@ -143,25 +143,33 @@ static AsyncRequestBody fromString(String string) {
     }
 
     /**
-     * Creates a {@link AsyncRequestBody} from a byte array. The contents of the byte array are copied so modifications to the
-     * original byte array are not reflected in the {@link AsyncRequestBody}.
+     * Creates a {@link AsyncRequestBody} from a byte array.
      *
      * @param bytes The bytes to send to the service.
      * @return AsyncRequestBody instance.
      */
     static AsyncRequestBody fromBytes(byte[] bytes) {
-        return new ByteArrayAsyncRequestBody(bytes, Mimetype.MIMETYPE_OCTET_STREAM);
+        return ByteBuffersAsyncRequestBody.from(bytes);
     }
```

Comment:
Similar to `SdkBytes`, can we keep the defensive copy here, but introduce a `fromBytesUnsafe` to create it without the defensive copy, so it's more obvious to the caller that they should not mutate the byte array after it is passed?